File: c/searching_algorithms/binary_search/binarySearch.c
#include <stdio.h>

int binarySearch(int element[],int size, int key){
    int low = 0;
    int high = size - 1;

    while(low<=high){
        int middle = low + (high - low) / 2;
        if(element[middle]==key){
            return middle;
        }else if(element[middle] < key){
            low = middle + 1;
        }else if(element[middle] > key){
            high = middle - 1;
        }else{
            return -1;
        };
    };
};

int main(){
    int element[] = {0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20}; // 
    int size = sizeof(element) / sizeof(element[0]);
    
    int key = 1;

    int result = binarySearch(element,size,key);
    printf("Found  %d at index %d\n",key,result);

    return 0;
};

File: python/searching_algorithms/binary_search/binarySearch.py
def binarySearch(array,key,low,high):
     # Repeat until the pointers low and high meet each other
    while low <= high:
       
        middle = low + (high - low) // 2

        if array[middle] == key:
            return middle
        elif array[middle] < key:
            low = middle + 1
        else:
            high = middle - 1

    return -1

array = [12,14,16,18,20,22,24]
key = 16

result = binarySearch(array,key,0,len(array) - 1)

if result != -1:
    print('Element is present at index: '+ str(result))
else:
    print("Not Found")